### PR TITLE
25 create searchfilter page searchbar

### LINKED ISSUE #25: Create : Search/Filter Page - SearchBar
## Summary

From authenticated user homepage, we will access to search/filter page. First thing to do is the search bar. Other components in this page will be created by other developers and we will bring them together afterwards.

Create a compact search bar, it should have a category selector dropdown which you can select from main categories, and an input field where the user can type whatever they want. 

No need to care about it much now but eventually, search must be instantenious. There will be a field under the search which another team will be building that renders item cards according to the search input. It should be case insensitive and item cards should be rendered on each button press.

Design and responsiveness must be readied with Tailwind, snapshot/unit tests should also be written.

### Screenshots

search bar

### Basic example

N/A

### Motivation

To be able to search through the items.

### PR DESCRIPTION
**Pull request by OmarBassatni97**

### CODE REVIEW

**Comment by OmarBassatni97** [comment]
I need your review guys

File: src/components/CustomButton.js
Comment (Iseluin) [inline src/components/CustomButton.js:10]: Great button !

File: src/components/SearchBar.js
Comment (Iseluin) [inline src/components/SearchBar.js:74]: Great job! On a second thought, since we are going to query the backend with onChange, we might not need a button here. Let's keep it for now.

**Review by Iseluin — APPROVED** [review]
💯 great job @OmarBassatni97

**Comment by Iseluin** [comment]
@OmarBassatni97 , you can merge this and delete your branch when you read my comments. Be sure to select "Squash and Merge" before you merge.

**Review by sarahaboudalal — APPROVED** [review]
It looks amazing Omar. Well done !!

**Pull request merged**
